credential/manifest/model.go

```diff
@@ -212,11 +224,80 @@ func IsValidCredentialApplicationForManifest(cm CredentialManifest, ca Credentia
 			if _, err := jsonpath.Compile(submissionDescriptor.Path); err != nil {
 				return fmt.Errorf("invalid json path: %s", submissionDescriptor.Path)
 			}
+		}
+
+		// index submission descriptors by id of the input descriptor
+		submissionDescriptorLookup := make(map[string]exchange.SubmissionDescriptor)
+		for _, d := range ca.PresentationSubmission.DescriptorMap {
+			submissionDescriptorLookup[d.ID] = d
+		}
+
+		// validate each input descriptor is fulfilled
+		for _, inputDescriptor := range cm.PresentationDefinition.InputDescriptors {
+			submissionDescriptor, ok := submissionDescriptorLookup[inputDescriptor.ID]
```

Comment:
at this point in time how will the fulfilled input descriptors have been marked as valid?

Reply:
well I think at this point we don't even know they are fulfilled (that the last part of the for loop)

This for loop
checks if the IDs match up
checks if the formats match up
makes sure we can resolve json path

Reply:
ahhh misunderstood this - makes sense now